### PR TITLE
Creating new Synchronizer - TimeSlice streamer

### PR DESCRIPTION
**Pull request by Martin-Molinero**
#### Description

- Creating new `Synchronizer` which will consolidate and combine `TimeSlice`
streaming for both live and backtests modes. Will live in front of the
`DataManager`
- `SubscriptionSynchronizer` will be the `FrontierTimeProvider` exposed,
and owned, by the new `Synchronizer`
- Updating existing tests

#### Related Issue

Closes https://github.com/QuantConnect/Lean/issues/2608
#### Motivation and Context

https://github.com/QuantConnect/Lean/projects/5
Both live and backtest modes will now share the same TimeSlice streaming loop.
#### Requires Documentation Change

N/A
#### How Has This Been Tested?

Unit tests, regression tests and live deployments
#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [ ] I have added tests to cover my changes. 
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description` or `feature-<issue#>-<description>`

### CODE REVIEW

File: Engine/DataFeeds/IDataFeedTimeProvider.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+{
+    /// <summary>
+    /// Reduced interface which exposes required <see cref="ITimeProvider"/> for <see cref="IDataFeed"/> implementations
+    /// </summary>
+    public interface IDataFeedTimeProvider
+    {
+        /// <summary>
+        /// Continuous UTC time provider
+        /// </summary>
+        ITimeProvider TimeProvider { get; }
```
Comment (mchandschuh) [inline Engine/DataFeeds/IDataFeedTimeProvider.cs]: interesting? who outside the DF would need access to this?

File: Engine/DataFeeds/LiveTradingDataFeed.cs [hunk 2]
```diff
@@ -495,13 +481,13 @@ private Subscription CreateUniverseSubscription(SubscriptionRequest request)
                             args.Action == NotifyCollectionChangedAction.Add ? args.NewItems :
                             args.Action == NotifyCollectionChangedAction.Remove ? args.OldItems : null;
 
-                        if (items == null || _frontierUtc == DateTime.MinValue) return;
+                        if (items == null || _frontierTimeProvider.GetUtcNow() == DateTime.MinValue) return;
```
Comment (mchandschuh) [inline Engine/DataFeeds/LiveTradingDataFeed.cs]: let's extract `_frontierTimeProvider.GetUtcNow()` into a local variable so we're not invoking it three times here

File: Engine/DataFeeds/IDataFeedTimeProvider.cs (hunk 1 above)
Comment (Martin-Molinero) [reply Engine/DataFeeds/IDataFeedTimeProvider.cs]: Nobody else, only the `DF` and today actually only the `LiveTradingDataFeed`. Sending it directly through the `IDataFeed.Initialize()`

File: Engine/DataFeeds/LiveTradingDataFeed.cs (hunk 2 above)
Comment (Martin-Molinero) [reply Engine/DataFeeds/LiveTradingDataFeed.cs]: Whops! Shame on me, I'll update it 👍

**Pull request merged**